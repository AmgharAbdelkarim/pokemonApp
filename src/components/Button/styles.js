import styled from 'styled-components';

export const ButtonContainer = styled.div`
  > button {
    padding: 8px 12px;
    margin: 0 2px 2px 0;
    border: 1px solid rgba(255, 255, 255, 0);
    border-radius: 10px;
    box-sizing: border-box;
    text-decoration: none;
    font-weight: 300;
    color: #04336b;
    text-shadow: 0 0.04em 0.04em rgba(0, 0, 0, 0.35);
    text-align: center;
    min-width: 75px;
    &:hover {
      border-color: rgba(255, 255, 255, 1);
    }
    &:disabled,
    &[disabled] {
      opacity: 0.65;
      cursor: not-allowed;
      &:hover {
        border-color: rgba(255, 255, 255, 0);
      }
    }
  }
`;
