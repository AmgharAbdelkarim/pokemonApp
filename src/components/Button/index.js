import React from 'react';
import {ButtonContainer} from 'components/Button/styles';

const Button = ({children, clickHandler, disabled}) => {
  return (
    <ButtonContainer>
      <button disabled={disabled} onClick={clickHandler}>
        {children}
      </button>
    </ButtonContainer>
  );
};

export default Button;
