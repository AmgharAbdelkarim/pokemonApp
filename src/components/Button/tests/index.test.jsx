import { cleanup, render } from "@testing-library/react";

import Button from "components/Button";

afterEach(cleanup);

describe("Button", () => {
  it("should Button render correctly", () => {
    const props = {
      clickHandler: jest.fn(),
      children: "fake button label",
      disabled: true,
    };
    const { asFragment, getByText } = render(<Button {...props} />);
    expect(getByText("fake button label")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it("should Button Be Disabled", () => {
    const props = {
      clickHandler: jest.fn(),
      children: "fake button label",
      disabled: true,
    };
    const { asFragment, getByText, getByRole } = render(<Button {...props} />);
    expect(getByText("fake button label")).toBeInTheDocument();
    const submitButton = getByRole("button", { name: /fake button label/ });

    expect(submitButton).toBeDisabled();
    expect(asFragment()).toMatchSnapshot();
  });
  it("should Button be Enabled", () => {
    const props = {
      clickHandler: jest.fn(),
      children: "fake button label",
      disabled: false,
    };
    const { asFragment, getByText, getByRole } = render(<Button {...props} />);
    expect(getByText("fake button label")).toBeInTheDocument();
    const submitButton = getByRole("button", { name: /fake button label/ });

    expect(submitButton).toBeEnabled();
    expect(asFragment()).toMatchSnapshot();
  });
});
