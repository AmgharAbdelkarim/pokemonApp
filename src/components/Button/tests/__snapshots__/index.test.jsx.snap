// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should Button Be Disabled 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jbmpvc"
  >
    <button
      disabled=""
    >
      fake button label
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button should Button be Enabled 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jbmpvc"
  >
    <button>
      fake button label
    </button>
  </div>
</DocumentFragment>
`;

exports[`Button should Button render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jbmpvc"
  >
    <button
      disabled=""
    >
      fake button label
    </button>
  </div>
</DocumentFragment>
`;
